components/nuki_lock/nuki_lock.cpp

```diff
@@ -134,35 +136,58 @@ void NukiLockComponent::update() {
 }
 
 void NukiLockComponent::control(const lock::LockCall &call) {
-    if (this->nukiLock_->isPairedWithLock()) {
-        auto state = *call.get_state();
-        uint8_t result;
+    if (!this->nukiLock_->isPairedWithLock()) {
```

Comment:
I took the liberty of refactoring this method a bit. Here we bail out early, to prevent nesting too many if's